apps/discord-bot/src/domains/mark-solution.ts

```diff
@@ -126,13 +126,20 @@ export async function checkIfCanMarkSolution(
 			PERMISSIONS_ALLOWED_TO_MARK_AS_SOLVED.some((permission) =>
 				userPermissions.has(permission),
 			);
+
+		const doesUserHaveThreadOverwritesPermissions = thread
+			.permissionsFor(guildMember)
+			.has('ManageThreads');
+
 		if (!doesUserHaveOverridePermissions) {
```

Comment:
Why are we nesting if statements here? Can we just put it as:

If(!overridePerms || !channelOverridePerms)